modules/aws/irsa/main.tf

```diff
@@ -53,11 +53,27 @@ resource "aws_iam_role" "this" {
 }
 
 resource "aws_iam_policy" "permissions" {
+  for_each = {
+    for s in ["policy-attachment"] :
+    s => s
+    if var.policy_json != ""
+  }
   name   = var.name
   policy = var.policy_json
 }
 
 resource "aws_iam_role_policy_attachment" "permissions" {
-  policy_arn = aws_iam_policy.permissions.arn
+  for_each = {
+    for s in ["policy-attachment"] :
+    s => s
+    if var.policy_json != ""
+  }
+  policy_arn = aws_iam_policy.permissions[0].arn
+  role       = aws_iam_role.this.name
+}
+
+resource "aws_iam_role_policy_attachment" "extra_policy_permissions" {
+  for_each   = var.extra_policy_permissions_arn
```

Comment:
Not familiar with how this works with sets. Will it have the same issues as count does?

Reply:
Yeah It might have. I will update it tomorrow.

Reply:
It will generate a map, not a list as count. The keys will be the ARNs which might not be that pleasant for the eye